### PR TITLE
options/ansi: Add basic support for complex arithmetic

### PR DESCRIPTION
**Pull request by nikstrmp**

### CODE REVIEW

**Comment by 64** [comment]
Thanks for the PR, as mentioned on Discord some testing would be nice to have.

**Comment by nikstrmp** [comment]
> Thanks for the PR, as mentioned on Discord some testing would be nice to have.

Wrote some basic tests, if someone could review them, it would be nice.

**Comment by nikstrmp** [comment]
I'm so sorry, I forgot to push a change that I had made. Thank you for you understanding.

**Comment by nikstrmp** [comment]
I think this is ready for review (for real this time)

**Comment by avdgrinten** [comment]
LGTM but the fix-up commits should be squashed.

File: options/ansi/generic/complex-stubs.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#include <complex.h>
+
+extern "C" {
```
Comment (no92) [inline options/ansi/generic/complex-stubs.cpp:3]: This extern seems unnecessary.

File: options/ansi/meson.build [hunk 2]
```diff
@@ -6,6 +6,7 @@ endif
 ansi_sources = files(
 	'generic/stdlib-stubs.cpp',
 	'generic/assert-stubs.cpp',
+	'generic/complex-stubs.cpp',
```
Comment (no92) [inline options/ansi/meson.build:9]: The `-stubs` suffix should be dropped IMO.

**Pull request merged**